--- ibus-chewing-1.4.14/src/MakerDialogUtil.c.orig	2014-12-15 15:06:27.000000000 +0100
+++ ibus-chewing-1.4.14/src/MakerDialogUtil.c	2016-04-03 10:57:28.094233446 +0200
@@ -120,7 +120,7 @@
 	g_snprintf(result, MAKER_DIALOG_VALUE_LENGTH, "%d", intValue);
 	break;
     case G_TYPE_STRING:
-	g_snprintf(result, MAKER_DIALOG_VALUE_LENGTH,
+	g_snprintf(result, MAKER_DIALOG_VALUE_LENGTH, "%s",
 		   g_value_get_string(value));
 	break;
     default:
--- ibus-chewing-1.4.14/src/GSettingsBackend.c.orig	2014-12-15 15:06:27.000000000 +0100
+++ ibus-chewing-1.4.14/src/GSettingsBackend.c	2016-04-03 11:04:06.900883376 +0200
@@ -19,6 +19,7 @@
  * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
  */
 
+#include <inttypes.h>
 #include <ibus.h>
 #include <glib.h>
 #include "MakerDialogUtil.h"
@@ -122,7 +123,7 @@
 	g_snprintf(result, MAKER_DIALOG_VALUE_LENGTH, "%u",
 		   g_variant_get_uint32(gVar));
     } else if (g_variant_type_is_subtype_of(gVType, G_VARIANT_TYPE_UINT64)) {
-	g_snprintf(result, MAKER_DIALOG_VALUE_LENGTH, "%lu",
+	g_snprintf(result, MAKER_DIALOG_VALUE_LENGTH, "%" PRIu64,
 		   g_variant_get_uint64(gVar));
     } else if (g_variant_type_is_subtype_of(gVType, G_VARIANT_TYPE_INT16)) {
 	g_snprintf(result, MAKER_DIALOG_VALUE_LENGTH, "%d",
@@ -131,10 +132,10 @@
 	g_snprintf(result, MAKER_DIALOG_VALUE_LENGTH, "%d",
 		   g_variant_get_int32(gVar));
     } else if (g_variant_type_is_subtype_of(gVType, G_VARIANT_TYPE_INT64)) {
-	g_snprintf(result, MAKER_DIALOG_VALUE_LENGTH, "%ld",
+	g_snprintf(result, MAKER_DIALOG_VALUE_LENGTH, "%" PRId64,
 		   g_variant_get_int64(gVar));
     } else if (g_variant_type_is_subtype_of(gVType, G_VARIANT_TYPE_STRING)) {
-	g_snprintf(result, MAKER_DIALOG_VALUE_LENGTH,
+	g_snprintf(result, MAKER_DIALOG_VALUE_LENGTH, "%s",
 		   g_variant_get_string(gVar, NULL));
     }
     return result;
